CONTRIBUTING.md

```diff
@@ -0,0 +1,303 @@
+- Reviewing changes from other contributors.
+
+The rest of this document is concerned with any changes impacting the `nickel`
+repository, which may or may not involve changing the source of Nickel.
+
+# Table of content
+
+1. [Preamble](#preamble)
+1. [Resources](#resources)
+1. [Setup a development environment](#set-up-a-development-environment)
+1. [How to submit changes](#how-to-submit-changes)
+
+# Preamble
+
+Before contributing any non trivial change to this repository, please first
+check for the existence of related work such as issues or open pull requests. If
+there are none, it's better to discuss the change you wish to make via an issue,
+by email, or using any other method with the maintainers of this repository
+(listed below) before actually submitting something.
+
+# Resources
+
+## Documentation
+
+The following resources are oriented toward Nickel users:
+
+- The [README](./README.md) and the [design rationale](./RATIONALE.md)
+- The [blog post serie][blog-serie] and the [release blog post][blog-release].
+- The [user manual][user-manual].
+
+For Nickel contributors (or aspiring contributors), the following technical
+documentation is relevant:
+
+- The [crate documentation][doc-crate].
+- The [RFCs][rfcs]. There is currently no well established process for RFCs, but as a
+  rule of thumb, impactful changes to the design or the implementation of the
+  language are technically discussed and documented in a dedicated RFC document.
+- The [technical notes][doc-notes]. Various notes gathering thoughts, proposals,
+  or issues about an aspect of the language at one point in time that the author
+  thought important to keep for posterity. They are usually more informal than
+  RFCs, of smaller scope, and their content may become obsolete more easily.
+
+## People
+
+Nickel is maintained by [Tweag][tweag]. The current team of maintainers is
+comprised of:
+
+- Yann Hamdaoui (@yannham)
+- François Caddet (@francoiscaddet)
+- Erin Van Der Veen (@...)
+- Steven Shaw (@...)
+
+You can find some of us on our [matrix channel][matrix-nickel] (and in
+particular the Devs room), or fire an email at `nickel-lang@protonmail.com`.
+
+# Set up a development environment
+
+There are two ways to set up a development environment for Nickel: using
+[Nix][nix], or directly via your preferred system package manager. Nix is able
+to drop you in a development shell with everything needed (the Rust toolchain,
+mostly) to hack on Nickel in one command, without installing anything globally
+on your system. _While Nix is also capable of building Nickel by itself, using
+cargo (either a system-wide installation or the one given by the Nix development
+shell) is the recommended way of building when working on the Nickel repository
+itself_. The reason is that incremental compilation for Rust and Nix is not
+there yet, and incremental rebuilds using only Nix are going to be painfully
+long.
```

Comment:
I think most `CONTRIBUTING.md` rather offload those instructions to a `HACKING.md` (or separate guide) document (Haskell, Rust, OCaml, etc.), that they link to in `CONTRIBUTING.md`. I may have started thinking that this would be very small, and thus could inline it, but the WASM stuff ended up being non-trivial, a separate guide may be a better approach.